### ISSUE TITLE
Missing warning/error when referring to event-driven var from summed var

### ISSUE THREAD

**Issue by mstimberg** [brian-team/brian2#1263]
In `Synapses`, we raise an error if a clock-driven variable refers to an event-driven variable. In contrast, we do not raise an error (or a warning) if a summed variable refers to an event-driven variable, even though this is probably a mistake as well. A summed variable gets evaluated at every time step and therefore all variables it refers to need to be updated at every time step as well.

The following is an example of an equation that should raise an error (or a warning?)
```Python
syn = Synapses(inp, out, '''ds/dt = -s/(3*ms) : 1 (event-driven)
                            x_post = s : 1 (summed)''', on_pre='s += 1')
```

**Comment by rohithvarma3000** [comment]
Hello @mstimberg,
I have gone through the code and found two locations where the check could be placed potentially. We could place it in the `check_flags` function under the Equations class or in the `__init__` method of the Synapses class. Since these flags may be used simultaneously by the Synapses objects only, I think it will be better to put in the `__init__` method of the Synapses class. What do you think?
Also, do you think we need to raise an error or a warning for this?

**Comment by mstimberg** [comment]
Yes, I agree that `Synapses.__init__` would be the best place. We'll have to make sure that the error is only raised when necessary, i.e. when the right-hand-side of the `summed` variable depends on an event-driven equation. Note that this could happen indirectly, like in the following (completely made up) example:
```Python
syn = Synapses(inp, out, '''ds/dt = -s/(3*ms) : 1 (event-driven)
                            t = s : 1
                            x_post = t : 1 (summed)''', on_pre='s += 1')
```
You should be able to use `Equations.get_substituted_expressions()` to deal with this issue. I don't really see a legitimate use case for the `summed` + `event-driven` combination, therefore I think that it will be used by mistake only. So I'd rather think it should be an error. Maybe @thesamovar has an opinion on this?

**Comment by mstimberg** [comment]
Closed via #1274
